src/util/locking/SingleThreadedResourceLocker.ts

```diff
@@ -25,13 +25,16 @@ export class SingleThreadedResourceLocker implements ResourceLocker {
    */
   public async acquire(identifier: ResourceIdentifier): Promise<Lock> {
     this.logger.verbose(`Acquiring lock for ${identifier.path}`);
-    return new Promise(async(resolve): Promise<Lock> =>
+    return new Promise((resolve, reject): void => {
       this.locks.acquire(identifier.path, (done): void => {
         this.logger.verbose(`Acquired lock for ${identifier.path}`);
-        resolve({ release: async(): Promise<void> => {
-          this.logger.verbose(`Released lock for ${identifier.path}`);
-          done();
-        } });
-      }));
+        resolve({
+          release: async(): Promise<void> => {
+            this.logger.verbose(`Released lock for ${identifier.path}`);
+            done();
+          },
+        });
+      }).catch(reject);
```

Comment:
Had a look into this since the linter is actually wrong here, but it's because of bad typings in the library we're using. If you call the `acquire` function of the library with as second parameter a function that has a `done` parameter, it does not return a promise as seen in the [docs](https://www.npmjs.com/package/async-lock). Unfortunately the typings of the library seem to indicate that the third parameter is not optional in that case (while it is in practice), perhaps due to problems with overlapping interfaces.

So the solution here would work since that catch is never going to fire anyway. The correct solution would be to add a callback to the acquire call so the linter sees it's not going to be a promise (but that would just be an empty callback since we don't use it), or change it to the Promise implementation of the library. But both of these would actually be more verbose than what it is now so I'll leave it up to you 😄

Reply:
But why are we using the callback-based interface in the first place? If it has promises, then we don't need the `new Promise` wrapper. I'll have a look.

Reply:
> If you call the `acquire` function of the library with as second parameter a function that has a `done` parameter, it does not return a promise as seen in the [docs](https://www.npmjs.com/package/async-lock).

Actually, that's not correct; it always returns a promise. So the above code is right.

Reply:
This is the TS interface of the acquire function of the library:
```typescript
    acquire<T>(key: string | string[],
               fn: (() => T | PromiseLike<T>) | ((done: AsyncLockDoneCallback<T>) => any),
               opts?: AsyncLockOptions): Promise<T>;
    acquire<T>(key: string | string[],
               fn: (done: AsyncLockDoneCallback<T>) => any,
               cb: AsyncLockDoneCallback<T>,
               opts?: AsyncLockOptions): void;
```
In practice we are using the second one but the linter assumes we're using the first one. So I don't see how it always returns a promise?

> But why are we using the callback-based interface in the first place?

They return a promise that resolves when the lock is released, not when it is acquired unfortunately.

Reply:
> So I don't see how it always returns a promise?

That was just judging from the docs, but given that it does not crash when calling `.catch`, it must be returning a promise 🙂